src/mobu/services/business/gitlfs.py

```diff
@@ -65,10 +75,23 @@ async def execute(self) -> None:
         created anew.
         """
         self.logger.info("Running Git-LFS check...")
-        with self.timings.start("execute git-lfs check") as sw:
-            await self._git_lfs_check()
-        elapsed = sw.elapsed.total_seconds()
-        self.logger.info(f"...Git-LFS check finished after {elapsed}s")
+        event = GitLfsCheck(
+            success=False,
+            **self.common_event_attrs(),
+        )
+        try:
+            with self.timings.start("execute git-lfs check") as sw:
+                await self._git_lfs_check(event)
+            elapsed = sw.elapsed.total_seconds()
+            self.logger.info(f"...Git-LFS check finished after {elapsed}s")
+
+            event.duration_total = sw.elapsed
+            event.success = True
+        except:
+            event.success = False
+            raise
+        finally:
+            await ed.events.git_lfs_check.publish(event)
```

Comment:
Is this directly-accessing-the-events-dependency-singleton pattern horrible? vs. threading it through the process context and injecting it as a parameter into the business constructor?

Reply:
Yeah, there are a few things that bother me about this: accessing the dependency directly, the renaming of imports (I don't like doing that unless one has to because multiple modules have symbols with the same name), and using a really short name that is kind of meaningless if one is looking at just a portion of the file. I tend to stick pretty closely to dependency injection so that it's part of the business as `self.events`, which is almost as short and I think somewhat clearer.

Reply:
[Here](https://github.com/lsst-sqre/mobu/pull/387/commits/bc86c8c379aade7608509d8aadef16aac652efa8) is this change to access it through the `ProcessContext`, I think that's the right place to put it in Mobu